src/torrent_peer.cpp

```diff
@@ -107,8 +107,8 @@ namespace libtorrent
 			if (e1 > e2) swap(e1, e2);
 			address_v6::bytes_type b1 = e1.address().to_v6().to_bytes();
 			address_v6::bytes_type b2 = e2.address().to_v6().to_bytes();
-			int mask = memcmp(&b1[0], &b2[0], 4) ? 0
-				: memcmp(&b1[0], &b2[0], 6) ? 1 : 2;
+			int mask = std::memcmp(&b1[0], &b2[0], 4) ? 0
```

Comment:
would you mind making `mask` const here? also, instead of `&b1[0]`, would it make more sense to use `b1.data()`?